sprocs/assessment_instances_insert.sql

```diff
@@ -26,19 +28,61 @@ BEGIN
 
     -- ######################################################################
     -- determine the "number" of the new assessment instance
+    IF group_work THEN
+        SELECT gu.group_id
+        INTO  groupId
+        FROM  group_users as gu
+        JOIN  groups AS g ON(g.id = gu.group_id)
+        JOIN  group_configs AS gc ON (gc.id = g.group_config_id)
+        WHERE gu.user_id = assessment_instances_insert.user_id AND
+        gc.assessment_id = assessment_instances_insert.assessment_id AND
+        gc.deleted_at IS NULL AND
+        g.deleted_at IS NULL;
+        
```

Comment:
Should we raise an exception here if `groupId` is NULL?

Please review other SQL in this PR and think about whether we need to have more defensive checks like this. We don't have to check everything, but some level of checking for consistency is good in cases where the DB structure does not enforce it.